release-notes.txt

```diff
@@ -1,3 +1,619 @@
+* sam0/usbdev: partial revert of #17086 (#19384)
+* sys/usbus: check for the number of required and provided EPs in
+  static configurations (#19371)
+
+Device Drivers
+--------------
+
++ black/white e-Paper/e-Ink display driver (#12509)
++ drivers/mfrc522: add new driver (#16782)
++ drivers/servo: reimplement with high level interface (#18392)
++ drivers/usbdev_synopsys_dwc2: add ESP32x power management (#19399)
++ drivers: add driver for ST VL6180X ranging and ambient light sensor (#10462)
+* drivers/at24cxxx: merge mtd driver with at24cxxx.c (#19281)
+* drivers/periph_common/flashpage: fix silent error (#19175)
+* drivers/slipdev: implement sleep states (#17810)
+* drivers/ws281x: improve timing for ESP32x (#19422)
+
+Documentation
+-------------
+
++ doc: Add out of tree RIOT app documentation (#19324)
+* blobs: include documentation in doxygen (#19200)
+* doc/porting-boards.md: fix code snippets in md view (#19206)
+* Doc: document generate-makefile.ci (#19239)
+* doccheck: avoid generating dot or html (#19228)
+
+Build System / Tooling
+----------------------
+
++ tools/compile_like_murdock: Add features (#19351)
++ tools/doccheck: add simple exclude to doccheck (#19240)
++ tools/renode: add support for target reset (#19375)
+* build system: Fix linker feature test with newlib 4.3.0 (#19232)
+* ci/print_versions: print all ESP32 toolchains (#19309)
+* dist/tools/doccheck: Fix grep warning (#19220)
+* dist/tools/insufficient_memory: Improve and clean up script (#19291)
+* esptools/install.sh: Fix shellcheck issues (#19325)
+* Fix `compile_like_murdock.py` when board is empty (#19370)
+* make: pass make jobserver to cargo (#19203)
+* Makefile.dep: pull in FEATURES_REQUIRED for periph (#19234)
+* toolchain/gnu.inc.mk: don't check version if CC is not installed (#19137)
+
+Kconfig
+-------
+
+* drivers/sx126x: make MODULE_SX126X_RF_SWITCH hidden in kconfig (#19194)
+* Fix SX126X-STM32WL module in Kconfig (#19191)
+
+Examples
+--------
+
++ examples/gnrc_border_router: add BLE as downlink option (#19321)
+
+Testing
+-------
+
++ bors.yaml: re-activate labels check + add block_labels (#19161)
++ CI: re-add "synchronize" event to check-labels (#19136)
++ gh-actions: report release-tests result to Matrix (#19102)
+* CI: .murdock: fix bors fast build (#19241)
+* CI: .murdock: use build checkout for ccache tmp (#19274)
+* CI: seperate check-labels and check-commits workflows (#19120)
+* CI: update check-labels-action (#19101)
+* test/periph_rtc: reset struct tm time between tests (#19344)
+* tests/gnrc_rpl: Disable CI test for native (#19171)
+* tests/periph_flashpage_unittest: enable periph_flashpage (#19215)
+
+API Changes
+-----------
+
++ usbdev: Add dedicated stall functions (#17086)
+* drivers/mtd_flashpage: implement pagewise API, don't use raw
+  addresses (#19258)
+* gcoap: Finish the gcoap_get_resource_list_tl ->
+  gcoap_get_resource_list renaming (#19295)
+* sys/flash_utils: helpers to store data in flash (#18148)
```

Comment:
Not sure *why* this is an API change

But the new servo API and the `ESP_WIFI_SSID` -> `WIFI_SSID` change sure is.

Reply:
The original PR was tagged as API Change (https://github.com/RIOT-OS/RIOT/pull/18148).
Is it intended? @maribu

Reply:
Yes, `phydat_unit_to_str()` is no longer thread safe on Harvard architectures and deprecated in favor of `phydat_unit_print()` and `phydate_unit_write()`, which are thread safe and, subjectively, easier to use.

It is a sublte API change likely nobody notices, as `phydat_unit_to_str()` was likely never used by applications directly, but rather indirect by including the `saul` shell command or by using `phydat_to_json`. Both have been updated to use the not-deprecated API. So I don't think that anyone would be affected.

Reply:
ok, I remove it from the API Changes then.